core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -116,6 +117,38 @@ public ThisT scanManifestsWith(ExecutorService executorService) {
     return self();
   }
 
+  @Override
+  public ThisT toBranch(String branch) {
+    throw new UnsupportedOperationException("Performing operations on a branch is currently not supported");
+  }
+
+  /***
+   * Will be used by snapshot producer operations to create a new ref if an invalid branch is passed
+   * @param branch ref name on which operation is to performed
+   */
+  protected void createNewRef(String branch) {
+    SnapshotRef branchRef = SnapshotRef.branchBuilder(this.current().currentSnapshot().snapshotId()).build();
+    TableMetadata.Builder updatedBuilder = TableMetadata.buildFrom(this.current());
+    updatedBuilder.setRef(branch, branchRef);
+    ops.commit(ops.current(), updatedBuilder.build());
+  }
+
+  /***
+   * A setter for the target branch on which snapshot producer operation should be performed
+   * @param branch to set as target branch
+   */
+  protected void setTargetBranch(String branch) {
+    this.targetBranch = branch;
+  }
+
+  /***
+   * A getter for the target branch on which snapshot producer operation should be performed
+   * @return target branch
+   */
+  protected String getTargetBranch() {
```

Comment:
Iceberg typically does not use `get` in method names because it is useless. Either it should be replaced by a more specific verb or it should be omitted.

Reply:
I see from #5010 that this is used in each child implementation to get the parent snapshot.

I think it would be better for maintainability and future implementations if the `apply` method were passed the parent snapshot.